tasks/puppet/src/main/java/com/walmartlabs/concord/plugins/puppet/model/cfg/PuppetConfiguration.java

```diff
@@ -347,11 +383,34 @@ public static <T extends PuppetConfiguration> T parseFromMap(Map<String, Object>
      * Gets the fields with the {@link JsonProperty} annotation from a class.
      *
      * @param clazz Class from which to get the fields
-     * @return Array of fields
+     * @return List of fields
      */
-    private static List<Field> getJsonPropertyFields(Class clazz) {
+    private static List<Field> getJsonPropertyFields(Class<?> clazz) {
 
         return Arrays.stream(clazz.getDeclaredFields())
-                .filter(e -> e.getAnnotation(JsonProperty.class) != null).collect(Collectors.toList());
+                .filter(e -> e.getAnnotation(JsonProperty.class) != null)
+                .toList();
+    }
+
+    public enum HttpVersion {
+        HTTP_1_1("http/1.1"),
+        HTTP_2("http/2.0"),
+        DEFAULT("default");
+
+        private final String value;
+
+        HttpVersion(String value) {
+            this.value = value;
+        }
+
+        public static HttpVersion from(String val) {
+            for (HttpVersion version : HttpVersion.values()) {
+                if (version.value.equals(val)) {
```

Comment:
toUpperCase? Seems like the http version string is now considered case-sensitive https://www.rfc-editor.org/rfc/rfc7230#page-80

Reply:
Fair enough. The input values aren't directly used any further than determining mapping to the correct enum. Doesn't hurt for it to match the spec since it's so close anyway.